api/analyzedphenotypes.ontology.api.inc

```diff
@@ -0,0 +1,391 @@
+}
+
+// * @TODO: leave out when api reorganize/update is complete and use functions below.
+function analyzedphenotypes_defaultontologyload() {
+  return ap_load_defaultontology();
+}
+
+
+///////
+
+
+/**
+ * FUNCTION:
+ * Insert into chado.cvterm or test existence of a cvterm.
+ * and return cvterm id to be set as default value in configuration page.
+ * Dependencies: ap_define_defaultontology().
+ *
+ * @return
+ *   An array, controlled vocabulary terms cvterm id.
+ *
+ * @see hook_install() implementation in install.inc.
+ */
+function ap_load_defaultontology() {
+  $ontology = ap_define_defaultontology();
+
+  // This array to hold default controlled vocabulary terms used in this system.
+  $ontology_load = array();
+
+  // Each controlled vocabulary - cv.
+  foreach($ontology['terms'] as $abbrev => $details) {
+    // Each term in a Controlled Vocabulary cv.
+    foreach($details['terms'] as $term) {
+      // Fetch the term, see if previously added.
+      $a = array(
+        'name' => $term['name'],
+        'cv_id' => array('name' => $details['name']),
+      );
+
+      if (function_exists('chado_get_cvterm')) {
+        $t = chado_get_cvterm($a);
+      }
+      else {
+        $t = tripal_get_cvterm($a);
+      }
+
+      if (!$t) {
+        // Term not in chado.cvterm. Insert!
+        // #1. Controlled vocabulary.
+        $ins_cv = tripal_insert_cv($details['name'], $details['definition']);
+
+        // #2. Terms.
+        $term['cv_name'] = $details['name'];
+        $ins_cvterm = tripal_insert_cvterm($term);
+      }
+
+      // CV TERM NAME.
+      $name = ($t) ? $t->name : $ins_cvterm->name;
+      // Match predefined variable to cvterm saved.
+      $map_term = array_search($name, $ontology['map_term']);
+
+      // Save term to be used as configuration page default settings.
+      // eg. ap_location = 1231.
+      $ontology_load[$map_term] = ($t) ? $t->cvterm_id : $ins_cvterm->cvterm_id;
+    }
+  }
+
+  return $ontology_load;
+}
+
+
+/**
+ * FUNCTION:
+ * Set ontology relationship of a trait in chado.cvterm_relationship.
+ * Dependencies: ap_get_variablenames() - systermvars.api.
+ *
+ * @param $identifiers
+ *   An array with the following keys:
+ *   - trait_cvterm_id : cvterm id number to which the ontology will be associated to.
+ *   and - ontology_cvterm_id : cvterm id number of ontology term that will be associated to a cvterm_id.
+ *
+ * @param $options
+ *   An array to be used as additional options to be performed to
+ *   returned result. Keys:
+ *   - replace : boolean true or false, when set to true attempt to replace existing ontolgy.
+ */
+function ap_set_ontology($identifiers, $options = array('replace' => FALSE)) {
+  $sysvar = ap_get_variablenames(
+    array('variablename' => 'related'),
+    array('set' => 'terms')
+  );
+
+  $sysvar_related = variable_get($sysvar);
+
+  $values = array(
+    'type_id' => $sysvar_related,
+    'object_id' => $identifiers['trait_cvtermid'],
```

Comment:
You should have checks at the beginning of the function that report a tripal error/warning if either of these are not set in $identifiers ;-)